src/gpu/program.rs

```diff
@@ -0,0 +1,80 @@
+use std::env;
+
+#[cfg(feature = "opencl")]
+use ec_gpu_gen::Limb64;
+use log::info;
+#[cfg(feature = "cuda")]
+use rust_gpu_tools::cuda;
+#[cfg(feature = "opencl")]
+use rust_gpu_tools::opencl;
+use rust_gpu_tools::{Device, Framework, GPUError as GpuToolsError, Program};
+
+#[cfg(not(all(feature = "cuda", feature = "opencl")))]
+use crate::gpu::error::GPUError;
+use crate::gpu::error::GPUResult;
+#[cfg(feature = "opencl")]
+use crate::gpu::sources;
+use crate::gpu::GpuEngine;
+use pairing::Engine;
+
+/// Returns the program for the preferred [`rust_gpu_tools::device::Framework`].
```

Comment:
this is very similar than the neptune version, should this be put into rust-gpu-tools in an abstract fashion?

Reply:
I had the same thought. The problem is that the part below (https://github.com/filecoin-project/bellperson/pull/218/files/4a577d76bb62c0338c54c408a3ee8eea17b4e1fb#diff-48d758a332b8b000b1386054877cf659fcc40ae7a2e4163552b0ad4659d2f1a9R60-R77) is slightly different (and will always differ between libraries (also the returned errors are different). Hence I'd keep that duplicated piece of code for now.